cmd/ndm_daemonset/filter/osdiskexcludefilter.go

```diff
@@ -130,8 +130,18 @@ func (odf *oSDiskExcludeFilter) Exclude(blockDevice *blockdevice.BlockDevice) bo
 			partitionRegex = "[0-9]*$"
 		}
 		regex := "^" + excludeDevPath + partitionRegex
-		klog.Infof("applying os-filter regex %s on %s", regex, blockDevice.DevPath)
-		if util.IsMatchRegex(regex, blockDevice.DevPath) {
+
+		// if the device is a dm device, need to check the /dev/mapper instead of /dev/dm-* because the
+		// mounts file will always have entry to the mapper path
+		var deviceMountPath string
+		if util.Contains(blockdevice.DeviceMapperDeviceTypes, blockDevice.DeviceAttributes.DeviceType) {
```

Comment:
should we do similar thing in the path filter also and filter the dm devices if provided?

Reply:
Done. 
In os disk filter also, we will only check for `dm-X` path. So other filters need not be changed.